### DISCUSSION TITLE
Nemo drag and drop with right click

### DISCUSSION THREAD

**Discussion by Qronikarz** [linuxmint/discussions#404] [Ideas]
On Windows it is possible to drag and drop files with right click instead of the left click. It allows you to open the context menu where you can choose which action do you want to perform - copy, move or make a shortcut.

I think it would be useful to have it since there are cases where you want to create a copy on the same drive, but also there are cases where you want to move the files to another drive and drag and drop by default assumes otherwise.

Here's animation of how that works:
https://www.cnet.com/tech/services-and-software/windows-right-click-move-files/

**Comment by mtwebster** [comment]
You can modify dnd behavior using modifier keys during the drag:

- No modifier: normal behavior - move for dnd on the same filesystem, copy otherwise.
- Control: Always copy
- Alt: Ask (context menu)
- Control-Shift: Link

I don't think we have enough control of the drag-and-drop interface we use to modify behavior depending on mouse button.

**Reply by Qronikarz** [reply]
> You can modify dnd behavior using modifier keys during the drag:

Bummer that it's only during the drag and drop. If it was before maybe I could at least make some macro. 

> Alt: Ask (context menu)

Can I make this default when doing the dnd from one disk to another disk?

> I don't think we have enough control of the drag-and-drop interface we use to modify behavior depending on mouse button.

That's also a bummer. Is it because the drag and drop mechanism has no way to know which button was used to start and it always assumes that left mouse click was used or because the context menu appears instantly as soon as you press the mouse button instead of when you release it?
